### PR TITLE
Add some tests for jruby/jruby

### LINKED ISSUE #652: test: test ComputeSourceDir / ComputeTestDir with jruby/jruby
The bug #650 was obtained when trying to fix a build from https://github.com/jruby/jruby. It looks like a project using heavily multi-modules.

We should test ComputeSourceDir and ComputeTestDir with their pom.xml file.

### PR DESCRIPTION
**Pull request by surli**
fix #652

### CODE REVIEW

**Comment by surli** [comment]
@fermadeiral @btellstrom I took the same pom.xml than those from jruby/jruby some I'm not checking really how it works with the properties to setup another source folder, but I think it's already checked in another test.

Then I check in an exhaustive way how it works from one module to its parent to another module to check the source and test directories. Tell me what you think about it.

File: repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputeSourceDir.java [hunk 1]
```diff
@@ -28,8 +28,12 @@
 import java.util.Optional;
 
 import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertThat;
 import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
```
Comment (fermadeiral) [inline repairnator/repairnator-pipeline/src/test/java/fr/inria/spirals/repairnator/process/step/paths/TestComputeSourceDir.java]: These imports are not used.

**Comment by fermadeiral** [comment]
Looks good to me. But we still miss tests on the `allModules` field. Maybe we could open a specific issue for that.

**Comment by surli** [comment]
>  But we still miss tests on the allModules field

My bad. I'll improve my tests

**Comment by btellstrom** [comment]
Looks good to me to, appreciate the comments as well.

**Comment by surli** [comment]
I just added the missing tests for allModules: there was indeed a bug with this one.
When calling the method with allModules from a module, it was getting the source (or test) dir from its parent, but not from the module of its parent. 
Now it's fixed.

**Comment by surli** [comment]
it's ready for merging @fermadeiral

**Comment by fermadeiral** [comment]
Oh, ok. I didn't know that I was supposed to merge.
Thank you @surli for the tests.

**Pull request merged**